src/services/node/NodeTransactionPoolService.ts

```diff
@@ -14,24 +14,172 @@
 // You should have received a copy of the GNU General Public License
 // along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+import { DispatchClass, Extrinsic, Weight } from '@polkadot/types/interfaces';
+import { u32 } from '@polkadot/types-codec';
+import BN from 'bn.js';
+
 import { INodeTransactionPool } from '../../types/responses';
 import { AbstractService } from '../AbstractService';
 
 export class NodeTransactionPoolService extends AbstractService {
-	async fetchTransactionPool(): Promise<INodeTransactionPool> {
+	/**
+	 * Fetch the transaction pool, and provide relevant extrinsic information.
+	 *
+	 * @param includeFee Whether or not to include the fee's and priority of a extrinsic
+	 * in the transaction pool.
+	 */
+	public async fetchTransactionPool(
+		includeFee: boolean
+	): Promise<INodeTransactionPool> {
 		const { api } = this;
-
 		const extrinsics = await api.rpc.author.pendingExtrinsics();
 
-		const pool = extrinsics.map((ext) => {
+		if (includeFee) {
+			const pool = await Promise.all(
+				extrinsics.map((ext) => this.extractExtrinsicInfo(ext))
+			);
+
 			return {
-				hash: ext.hash.toHex(),
-				encodedExtrinsic: ext.toHex(),
+				pool,
 			};
-		});
+		} else {
+			return {
+				pool: extrinsics.map((ext) => {
+					return {
+						hash: ext.hash.toHex(),
+						encodedExtrinsic: ext.toHex(),
+					};
+				}),
+			};
+		}
+	}
+
+	/**
+	 * Extract all information related to the extrinsic, and compute it's
+	 * priority in the transaction pool.
+	 *
+	 * @param ext Extrinsic we want to provide all the information for.
+	 */
+	private async extractExtrinsicInfo(ext: Extrinsic) {
+		const { api } = this;
+		const { hash, tip } = ext;
+		const {
+			class: c,
+			partialFee,
+			weight,
+		} = await api.rpc.payment.queryInfo(ext.toHex());
+		const priority = await this.computeExtPriority(ext, c, weight);
 
 		return {
-			pool,
+			hash: hash.toHex(),
+			encodedExtrinsic: ext.toHex(),
+			tip: tip.toString(),
+			priority: priority,
+			partialFee: partialFee,
 		};
 	}
+
+	/**
+	 * We calculate the priority of an extrinsic in the transaction pool depending
+	 * on its dispatch class, ie. 'normal', 'operational', 'mandatory'.
+	 *
+	 * The following formula can summarize the below logic.
+	 * tip * (max_block_{weight|length} / bounded_{weight|length})
+	 *
+	 * Please reference this link for more information
+	 * ref: https://github.com/paritytech/substrate/blob/fe5bf49290d166b9552f65e751d46ec592173ebd/frame/transaction-payment/src/lib.rs#L610
+	 *
+	 * @param ext
+	 * @param c
+	 * @param weight
+	 */
+	private async computeExtPriority(
+		ext: Extrinsic,
+		dispatchClass: DispatchClass,
+		weight: Weight
+	): Promise<string> {
+		const { api } = this;
+		const { tip, encodedLength: len } = ext;
+		const BN_ONE = new BN(1);
+		const sanitizedClass = this.defineDispatchClassType(dispatchClass);
+
+		const { maxBlock } = this.api.consts.system.blockWeights;
```

Comment:
```suggestion
		const { maxBlockWeight } = this.api.consts.system.blockWeights;
```

Reply:
So we can't actually rename this using this syntax as it's destructuring the key `maxBlock` from the object `blockWeights`, but I do agree that some explicit naming would be better here so I'll give it a change.

Reply:
changed